### PR TITLE
[Pickers] Edge IME Compatibility & IE demo page bug fix

### PR DESCRIPTION
**Pull request by andrewhwng**
#### Pull request checklist

- [x] Include a change request file using `$ yarn change`

#### Description of changes

1. There was a bug where typing inside the picker with a language requiring a composition state for typing characters (ie. Japanese) would not behave as expected in Edge. Specifically...
    - Typing a character that matches a suggestion will fill the text input to match the first suggestion, but the highlight does not occur
2. Demo page selector did not work in IE in most recent build when I was testing. Turns out it was caused due to the RegExp constructor not being supported in IE _iff_ there is a regexp passed in. It only takes a string...

#### Focus areas to test

- Any demo page in IE
- Pickers while testing composition based languages. To test, I will push another commit to add a Japanese string to the example page and provide more details on how to test it in a comment :) I'll be sure to remove the commit before merging.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/10610)

### CODE REVIEW

File: packages/tsx-editor/src/transpiler/exampleParser.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export function tryParseExample(example: string, supportedPackages: IBasicPackag
 
 /** @internal */
 export function _tryParseExample(example: string, supportedPackages: string[]): IExampleInfo | string {
-  const possibleComponents = example.match(new RegExp(COMPONENT_NAME_REGEX, 'gm'));
+  const possibleComponents = example.match(new RegExp(COMPONENT_NAME_REGEX.source, 'gm'));
```
Comment (ecraig12345) [inline packages/tsx-editor/src/transpiler/exampleParser.ts]: Thanks for noticing this! Sorry about the extra trouble but would you mind putting this in a separate PR (or I can)? I'm not familiar enough with Autofill to be comfortable approving that part of the change, but this part is fine and should really go in ASAP.

Comment (andrewhwng) [reply packages/tsx-editor/src/transpiler/exampleParser.ts]: Sure! just created [PR 10611](https://github.com/OfficeDev/office-ui-fabric-react/pull/10611), I'll get rid of these changes here

**Comment by andrewhwng** [comment]
**Testing**
In the most recent commit (in which I will delete before merging), I pushed an option in the demo page to include 'あいうえお' as an auto-fill option for the picker.

To get the typing to work, install Japanese on your OS, switch your typing language, and the first character can be matched by pressing the 'a' key (you'll need to then press enter to finish the composition). Auto-fill should work as expected.

Exporting to codepen will load the broken version and you can run the same test there to see the bug :)

**Comment by ecraig12345** [comment]
> In the most recent commit (in which I will delete before merging)...

Added do not merge label to make that clear. :) Feel free to remove it when you're ready.

File: packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx [hunk 2]
```diff
@@ -23,7 +23,8 @@ const _testTags: ITag[] = [
   'rose',
   'violet',
   'white',
-  'yellow'
+  'yellow',
+  'あいうえお'
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx]: DId you mean to add this here?

File: packages/office-ui-fabric-react/src/components/Autofill/Autofill.tsx [hunk 3]
```diff
@@ -217,8 +217,9 @@ export class Autofill extends BaseComponent<IAutofillProps, IAutofillState> impl
   private _onInputChanged = (ev: React.FormEvent<HTMLElement>) => {
     const value: string = this._getCurrentInputValue(ev);
 
-    // Right now typing does not have isComposing, once that has been fixed any should be removed.
-    this._tryEnableAutofill(value, this._value, (ev.nativeEvent as any).isComposing);
+    if (!this._isComposing) {
+      this._tryEnableAutofill(value, this._value, (ev.nativeEvent as any).isComposing);
```
Comment (joschect) [inline packages/office-ui-fabric-react/src/components/Autofill/Autofill.tsx]: Can we add a test? To catch some of these IME bugs?

File: packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx (hunk 2 above)
Comment (joschect) [reply packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx]: Should have read your comment. Ignore this.

Comment (andrewhwng) [reply packages/office-ui-fabric-react/src/components/pickers/examples/TagPicker.Basic.Example.tsx]: This was to have a deploy that's testable, I will delete it before merging! Hence the 'Do not merge' label on the PR now

File: packages/office-ui-fabric-react/src/components/Autofill/Autofill.tsx (hunk 3 above)
Comment (andrewhwng) [reply packages/office-ui-fabric-react/src/components/Autofill/Autofill.tsx]: I've just added a simple autofill test for composable languages :)

**Comment by andrewhwng** [comment]
@joschect tagging one last time for sanity checks :)

**Pull request merged**